lib/routes/account_required_actions.dart

```diff
@@ -44,60 +39,61 @@ class AccountRequiredActionsIndicator extends StatefulWidget {
 
 class AccountRequiredActionsIndicatorState
     extends State<AccountRequiredActionsIndicator> {
-  StreamSubscription<void> _promptEnableSubscription;
-  StreamSubscription<BackupSettings> _settingsSubscription;
-  BackupSettings _currentSettings;
+  StreamSubscription<dynamic> _promptBackupSubscription;
   bool showingBackupDialog = false;
-  bool _init = false;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    if (!_init) {
-      _settingsSubscription = widget._backupBloc.backupSettingsStream
-          .listen((settings) => _currentSettings = settings);
-
-      _promptEnableSubscription = widget._backupBloc.promptBackupStream
-          .delay(const Duration(seconds: 4))
-          .listen((needSignIn) async {
-        if (_currentSettings.promptOnError && !showingBackupDialog) {
-          showingBackupDialog = true;
-          widget._backupBloc.backupPromptVisibleSink.add(true);
-          popFlushbars(context);
-          showDialog(
-              useRootNavigator: false,
-              barrierDismissible: false,
-              context: context,
-              builder: (_) => EnableBackupDialog(
-                    context,
-                    widget._backupBloc,
-                    signInNeeded: needSignIn,
-                  )).then((_) {
-            showingBackupDialog = false;
-            widget._backupBloc.backupPromptVisibleSink.add(false);
-          });
-        }
-      });
-      _init = true;
-    }
+    final backupBloc = AppBlocsProvider.of<BackupBloc>(context);
+    _promptBackupSubscription?.cancel();
+    _promptBackupSubscription = Rx.combineLatestList([
+      backupBloc.backupSettingsStream,
+      backupBloc.promptBackupStream,
+      backupBloc.promptBackupDismissedStream,
```

Comment:
It seems that we can encapsulate the promptBackupDismissedStream and not handle it in the UI.
How about only push to the promptBackupStream if really prompt is needed?